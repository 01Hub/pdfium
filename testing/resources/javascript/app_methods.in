{{header}}
{{object 1 0}} <<
  /Type /Catalog
  /Pages 2 0 R
  /OpenAction 10 0 R
>>
endobj
{{object 2 0}} <<
  /Type /Pages
  /Count 1
  /Kids [
    3 0 R
  ]
>>
endobj
% Page number 0.
{{object 3 0}} <<
  /Type /Page
  /Parent 2 0 R
  /MediaBox [0 0 612 792]
>>
% OpenAction action
{{object 10 0}} <<
  /Type /Action
  /S /JavaScript
  /JS 11 0 R
>>
endobj
% JS program to exexute
{{object 11 0}} <<
  {{streamlen}}
>>
stream

app.alert("message", 1, 2, "title");
app.alert({"cMsg": "message", "cTitle": "title"});
app.alert({"cMsg": "message", "cTitle": "title", "nIcon": 3, "nType": 4});
app.alert(undefined);
app.alert(null);
app.alert(true);
app.alert(false);
app.alert(42);
app.alert([1, 2, 3]);
app.alert([1, 2, {"color": "red"}]);
app.alert({"color": "red"}, 5, 6, "title");
try {
  app.alert();
} catch (e) {
  app.alert("Caught expected error " + e);
}
try {
  app.alert({});
} catch (e) {
  app.alert("Caught expected error " + e);
}
try {
  app.alert({"color": "red", "size": 42});
} catch (e) {
  app.alert("Caught expected error " + e);
}

app.beep(1);

app.mailMsg(true);
app.mailMsg(false, "user@example.com");
app.mailMsg(false, "user@example.com", "cc@example.com",
            "bcc@example.com", "subject", "body");
app.mailMsg({"bUI": true});
app.mailMsg({"bUI": false, "cTo": "user@example.com"});
app.mailMsg({"bUI": false,
             "cTo": "user@example.com",
             "cCc": "cc@example.com",
             "cBcc": "bcc@example.com",
             "cSubject": "subject",
             "cMsg": "body"});
try {
  app.mailMsg();
} catch (e) {
  app.alert("Caught expected error " + e);
}
try {
  app.mailMsg(false);
} catch (e) {
  app.alert("Caught expected error " + e);
}
try {
  app.mailMsg({"color": "red", "size": 42});
} catch (e) {
  app.alert("Caught expected error " + e);
}

var result;
try {
  result = app.response("question");
  app.alert("result: " + result);
  result = app.response("question", "title", "default", true, "label");
  app.alert("result: " + result);
  result = app.response({"cQuestion": "question"});
  app.alert("result: " + result);
  result = app.response({
    "cQuestion": "question",
    "cTitle": "title",
    "cDefault": "default",
    "bPassword": true,
    "cLabel": "label"
  });
  app.alert("result: " + result);
} catch (e) {
  app.alert("unexpected error " + e);
}
try {
  app.response();
  app.alert("unexpected success");
} catch (e) {
  app.alert("Caught expected error " + e);
}
try {
  app.response({});
  app.alert("unexpected success");
} catch (e) {
  app.alert("Caught expected error " + e);
}

endstream
endobj
{{xref}}
{{trailer}}
{{startxref}}
%%EOF
