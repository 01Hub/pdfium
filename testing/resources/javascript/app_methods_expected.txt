title[icon=1,type=2]: message
title: message
title[icon=3,type=4]: message
Alert: undefined
Alert: null
Alert: true
Alert: false
Alert: 42
Alert: [1, 2, 3]
Alert: [1, 2, [object Object]]
title[icon=5,type=6]: [object Object]
Alert: Caught expected error app.alert: Incorrect number of parameters passed to function.
Alert: Caught expected error app.alert: Incorrect number of parameters passed to function.
Alert: Caught expected error app.alert: Incorrect number of parameters passed to function.
BEEP!!! 1
Mail Msg: 1, to=, cc=, bcc=, subject=, body=
Mail Msg: 0, to=user@example.com, cc=, bcc=, subject=, body=
Mail Msg: 0, to=user@example.com, cc=cc@example.com, bcc=bcc@example.com, subject=subject, body=body
Mail Msg: 1, to=, cc=, bcc=, subject=, body=
Mail Msg: 0, to=user@example.com, cc=, bcc=, subject=, body=
Mail Msg: 0, to=user@example.com, cc=cc@example.com, bcc=bcc@example.com, subject=subject, body=body
Alert: Caught expected error app.mailMsg: Incorrect number of parameters passed to function.
Alert: Caught expected error app.mailMsg: Incorrect number of parameters passed to function.
Alert: Caught expected error app.mailMsg: Incorrect number of parameters passed to function.
PDF: question, defaultValue=, label=, isPassword=0, length=2048
Alert: result: No
title: question, defaultValue=default, label=label, isPassword=1, length=2048
Alert: result: No
PDF: question, defaultValue=, label=, isPassword=0, length=2048
Alert: result: No
title: question, defaultValue=default, label=label, isPassword=1, length=2048
Alert: result: No
Alert: Caught expected error app.response: Incorrect number of parameters passed to function.
Alert: Caught expected error app.response: Incorrect number of parameters passed to function.
